### DISCUSSION TITLE
Issue importing prefixes data from csv

### DISCUSSION THREAD

**Discussion by F3d0s** [netbox-community/netbox#12241] [Q&A]
I am migrating some data from another Netbox environment, everything is going more or less well but I have an issue importing the Prefixes.

The problem is that when I import the data from the CSV file, it gives me an error on the `vlan` column and I get the error message: ***Record 3 vlan: Object not found: VlanName (2)***.
The field options indicate that the accessor of the `vlan` field is `vid`, so, in this case, it should be just 2, unfortunately, even if I just enter `2`, I get the same error

As an example, this is a part of the CSV I am trying to import:

```csv
prefix,status,vrf,site,vlan,role,description,vlan_group,pool,tags,depth
10.28.0.0/22,active,Global,Site1,VlanName (2),Clients,Clients D,VlanGroup1,FALSE,tag1,2
```

I wonder if I'm doing something wrong or if it's a bug because even after following the import instructions, it won't let me.

**Comment by mtinberg** [comment]
I don't think those are the right column names for import, it's is_pool and not just pool and depth is not a column for import.  For the VLAN association to work it needs to be able to uniquely identify the record, VLAN ID is not unique unless it's part of a VLAN Group or Site

eg.

prefix,status,vrf,site,vlan_group,vlan,description,comments
192.0.2.64/30,active,,,LAB,2,TEST-VLAN,A test
Field Options
Field   Required        Accessor        Description
prefix          —       IPv4 or IPv6 network with mask
vrf     —       name    Assigned VRF
tenant  —       name    Assigned tenant
site    —       name    Assigned site
vlan_group      —       name    VLAN's group (if any)
vlan    —       vid     Assigned VLAN
status          —
Operational status
role    —       name    Functional role
is_pool         —       —       All IP addresses within this prefix are considered usable
Specify "true" or "false"
mark_utilized   —       —       Treat as 100% utilized
Specify "true" or "false"
description     —       —       Description
comments        —       —       Comments
tags    —       slug    Tag slugs separated by commas, encased with double quotes (e.g. "tag1,tag2,tag3")
id      —       —       Numeric ID of an existing object to update (if not creating a new object)



—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: F3d0s ***@***.***>
Sent: Thursday, April 13, 2023 8:57 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Issue importing prefixes data from csv (Discussion #12241)


I am migrating some data from another Netbox environment, everything is going more or less well but I have an issue importing the Prefixes.

The problem is that when I import the data from the CSV file, it gives me an error on the vlan column and I get the error message: Record 3 vlan: Object not found: VlanName (2).
The field options indicate that the accessor of the vlan field is vid, so, in this case, it should be just 2, unfortunately, even if I just enter 2, I get the same error

As an example, this is a part of the CSV I am trying to import:

prefix,status,vrf,site,vlan,role,description,vlan_group,pool,tags,depth
10.28.0.0/22,active,Global,Site1,VlanName (2),Clients,Clients D,VlanGroup1,FALSE,tag1,2


I wonder if I'm doing something wrong or if it's a bug because even after following the import instructions, it won't let me.
